grai-integrations/source-bigquery/src/grai_source_bigquery/models.py

```diff
@@ -155,6 +156,16 @@ class Edge(BaseModel):
     constraint_type: Constraint
     metadata: Optional[Dict] = None
 
+    def __hash__(self):
+        """
+
+        Returns:
+
+        Raises:
+
+        """
+        return hash((self.source.full_name, self.destination.full_name, self.constraint_type))
```

Comment:
I might be missing it but why include the constraint_type in the hash? From the perspective of the Edge I think it's unique on 

```python
hash(((self.source.namespace, self.source.full_name), (self.destination.namespace, self.destination.full_name)))
```